tests/Doctrine/Tests/DBAL/Driver/StatementIteratorTest.php

```diff
@@ -2,39 +2,94 @@
 
 namespace Doctrine\Tests\DBAL\Driver;
 
+use Doctrine\DBAL\Driver\IBMDB2\DB2Statement;
+use Doctrine\DBAL\Driver\Mysqli\MysqliStatement;
+use Doctrine\DBAL\Driver\OCI8\OCI8Statement;
+use Doctrine\DBAL\Driver\SQLAnywhere\SQLAnywhereStatement;
+use Doctrine\DBAL\Driver\SQLSrv\SQLSrvStatement;
 use Doctrine\DBAL\Driver\Statement;
 use Doctrine\DBAL\Driver\StatementIterator;
+use Doctrine\DBAL\Portability\Statement as PortabilityStatement;
+use IteratorAggregate;
+use PHPUnit\Framework\MockObject\MockObject;
+use Traversable;
+use function extension_loaded;
 
 class StatementIteratorTest extends \Doctrine\Tests\DbalTestCase
 {
-    public function testGettingIteratorDoesNotCallFetch()
+    /**
+     * @dataProvider statementProvider()
+     */
+    public function testGettingIteratorDoesNotCallFetch(string $class)
     {
-        $stmt = $this->createMock(Statement::class);
+        /** @var IteratorAggregate|MockObject $stmt */
+        $stmt = $this->createPartialMock($class, ['fetch', 'fetchAll', 'fetchColumn']);
         $stmt->expects($this->never())->method('fetch');
         $stmt->expects($this->never())->method('fetchAll');
         $stmt->expects($this->never())->method('fetchColumn');
 
+        $stmt->getIterator();
+    }
+
+    public function testIteratorIterationCallsFetchOncePerStep()
```

Comment:
can haz void?

Reply:
Would it make sense to require a non-empty return type on test methods? It's `void` in 99% of cases, but if a test returns a dependency, it's in most cases of a known type.

Reply:
I don't understand the "non-empty return type" part. You mean making all tests return something specific, instead of void?

Reply:
Not sure it could be a requirement project-wide. For instance, `fetch()` could return either a row (`array`) or one field (`string|null`) or `false` (end of set). You'll have to have a very strict API to enable this, can't see it happening in the near future.